src/reconstruction/modelGeneration/modelVerification/verifyRuleSyntax.m

```diff
@@ -0,0 +1,44 @@
+function tf = verifyRuleSyntax(ruleString)
+% try to verify if a given rule string is syntactically correct
+% USAGE:
+%    tf = verifyRuleSyntax(ruleString)
+%
+% INPUT:
+%    ruleString:    The string in rule format (i.e. x(1) | x(2) & x(3) ...)
+%um
+% OUTPUT:
+%    tf:            Whether the given string is evaluateable (i.e. the
+%                   syntax is correct
+persistent x
+
+if isempty(x)
+    % 10000 genes should be large enough for most models.
+    x = true(10000,1); 
+end
+
+if isempty(ruleString)
+    tf = true;
+else
+    try
+        temp = eval(ruleString);
+        % if we reach this point it is fine
+        tf = true;
```

Comment:
`tf = true` could be set as default, no?

Reply:
No, rather false.
But I just noticed, that this is still bugged. Need to fix the second try/catch block..

Reply:
Yeah, doesn't matter 😄